### PR TITLE
Add NPM proxy cache

### PR DESCRIPTION
**Pull request by pecigonzalo**
Related https://github.com/sourcegraph/infrastructure/pull/2304

### CODE REVIEW

File: .buildkite/hooks/post-checkout [hunk 1]
```diff
@@ -5,3 +5,8 @@
 echo "Running git fetch..."
 git fetch
 echo "Running git fetch... done"
+
+echo "Setup NPM proxy..."
+yarn config set httpsProxy http://npm-proxy-cache:8080/
```
Comment (eseliger) [inline .buildkite/hooks/post-checkout:10]: is yarn not complaining about protocol mismatch here?

**Review by eseliger — APPROVED** [review]
LGTM

**Comment by eseliger** [comment]
Is this already deployed? Doesn't seem faster 🤔

**Comment by pecigonzalo** [comment]
No its not, I have to deploy sourcegraph/infrastructure#2304

File: .buildkite/hooks/post-checkout (hunk 1 above)
Comment (pecigonzalo) [reply .buildkite/hooks/post-checkout:10]: It did not, and I ran it locally and it threw a NPM warning but nothing else.

**Review by daxmc99 — APPROVED** [review]

**Comment by pecigonzalo** [comment]
Closing this PR and ill revert the other one. We dont benefit from `npm-cache` because we already keep a hot cache on each buildkite agent as they are not ephemeral. The only benefit here would be that the first run of an agent is faster, but that is negligible.

**Pull request closed without merging**